src/diff_edit.rs

```diff
@@ -457,5 +468,6 @@ fn _get_editor_from_settings_helper(
             default_editor
         }
     };
-    Ok(get_tool_config(ui.settings(), &editor_name)?)
+    let editor_config = get_tool_config(ui.settings(), &editor_name)?;
```

Comment:
would it be cleaner to make the caller do this, renaming this function to just `_get_editor_name_from_settings`?

Reply:
Good point, thanks for looking at it with fresh eyes! This should remove most of the ugliness.